index.html

```diff
@@ -81,235 +81,215 @@
+      removed, or its value is set to an empty string.</p>
+      <p>To <dfn>obtain the preload resource</dfn>, the user agent must run the
+      following steps:</p>
+      <ol>
+        <li>If the `href` attribute's value is the empty string, then abort
+        these steps.</li>
+        <li>[Resolve] the [URL] (<i>absolute url</i>) given by the `href`
+        attribute, relative to the element.</li>
+        <li>If the previous step fails, then abort these steps.</li>
+        <li>Validate the <i>request type</i> given by the `as` attribute. If
+        the attribute is omitted, or the provided value is not a [valid request
+        type], then initialize it to the empty string. [[!FETCH]]</li>
+        <li>Do a potentiall CORS-enabled fetch of the resulting <i>absolute
+        URL</i>, with the <i>mode</i> being the current state of the element's
+        [crossorigin] content attribute, the <i>origin</i> being the [origin]
+        of the [link] element's [node document], <i>type</i> set to <i>request
+        type</i>, and the <i>default origin behaviour</i> set to
+        <i>taint</i>.</li>
+      </ol>
+      <p>The <a>preload link</a> element MUST NOT [delay the load event] of the
+      element's [node document].</p>
+      <p class="note">The fetch initiated by a <a>preload link</a> is subject
+      to relevant CSP policies determined by the value of the `as` attribute -
+      e.g. when set to `image` the fetch is subject to `image-src`. Similarly,
+      the value of the `as` attribute is used to initialize request headers and
+      priority. If the `as` attribute is omitted, the request will be
+      initialized with similar processing and request properties as one
+      initiated by `XMLHttpRequest`.</p>
+      <p>Once a <dfn>preload resource has been <a data-lt=
+      "obtain the preload resource">obtained</a></dfn>, the user agent must run
+      these steps:</p>
+      <ol>
+        <li>If the load was successful, [queue a task] to [fire a simple event]
+        named `load` at the [link] element. Otherwise, [queue a task] to [fire
+        a simple event] named `error` at the [link] element.</li>
+        <li>Add request to preload response cache (TODO).
+          <p class="issue">TODO: define preload "response cache". Conceptually
+          this should probably reuse [SW's cache
+          logic](https://slightlyoff.github.io/ServiceWorker/spec/service_worker/#cache-objects)?
+          In effect, the preload cache is a UA maintained response cache with a
+          few interesting lifetime + eviction quirks.</p>
+        </li>
+      </ol>
+      <p>The user agent MUST NOT automatically execute or apply the resource
+      against the current page context, and the user agent MUST retain the
+      fetched response until it is <a data-lt=
+      "match a preloaded response">matched with another request</a>, or is no
+      longer valid.</p>
```

Comment:
is "no longer valid" defined somewhere? Will it be defined as part of the preload cache?